torch/distributed/optim/named_optimizer.py

```diff
@@ -260,11 +260,28 @@ def load_state_dict(self, state_dict: Mapping[str, Any]) -> None:
 
         self._optimizer.load_state_dict(new_state_dict)
 
-    # pyre-ignore [2]
-    def add_param_group(self, param_group: Any) -> None:
-        raise NotImplementedError(
-            "add_param_group not supported yet and might be implemented soon."
+    def add_param_group(self, param_group: Mapping[str, Any]) -> None:
+        warnings.warn(
+            "This API is still in develop and it might not work as expected "
+            "under the distributed training scenario."
```

Comment:
Curious what limitations the API might currently have? If there are clear issues with distributed training, we should highlight them. If not, it's probably sufficient to add to the docstring "This API is experimental and subject to change".

Reply:
I see. I will update the docstring, I guess my concern here is for the SPMD style training, there could be some corner cases we don't support, for example, what if user pass in different param groups on different ranks because each rank store different param?